Sources/SwiftParser/Directives.swift

```diff
@@ -106,29 +109,56 @@ extension Parser {
       while let poundIfHandle = firstIteration ? self.canRecoverTo(.poundIfKeyword) : self.canRecoverTo(anyIn: IfConfigContinuationClauseStartKeyword.self)?.handle,
         loopProgress.evaluate(self.currentToken)
       {
-        var (unexpectedBeforePoundIf, poundIf) = self.eat(poundIfHandle)
-        firstIteration = false
-        // Parse the condition.
+        var unexpectedBeforePound: RawUnexpectedNodesSyntax?
+        var pound: RawTokenSyntax
         let condition: RawExprSyntax?
-        switch poundIf.tokenKind {
-        case .poundIfKeyword, .poundElseifKeyword:
+        var atElifTypo: Bool {
+          guard self.at(TokenSpec(.pound)), self.currentToken.trailingTriviaText.isEmpty else {
+            return false
+          }
+          let identifierSpec = TokenSpec(.identifier, allowAtStartOfLine: false)
+          var lookahead = self.lookahead()
+          lookahead.consumeAnyToken()
+          guard lookahead.at(identifierSpec), lookahead.currentToken.tokenText == "elif", lookahead.currentToken.leadingTriviaText.isEmpty else {
+            return false
+          }
+          lookahead.consumeAnyToken()
+          return lookahead.at(identifierSpec)
+        }
+
+        if atElifTypo {
```

Comment:
I have just been thinking: You should be able to write this in terms of `consume(if:followedBy:)`. That way you wouldn’t need the `atElifTypo` variable. Something like (written in GitHub, so might have some syntax errors):

```swift
if let (pound, identifier) = self.consume(if: { $0.tokenKind == .pound && $0.trailingTrivia.isEmpty }, followedBy: { $0.tokenKind == .identifier && $0.tokenText == "elif" && $0.leadingTrivia.isEmpty })
```

Reply:
I managed to add the following: 
```swift
        if let (_pound, identifier) = self.consume(
          if: {
            $0.rawTokenKind == .pound && $0.trailingTriviaText.isEmpty
          }, followedBy: {
            $0.rawTokenKind == .identifier && $0.tokenText == "elif" && $0.leadingTriviaText.isEmpty
          }) { 
            // ...
          }
```

The problem is that we still use the `atElifTypo` in the while loop on line 149, for parsing the elements in between.

Reply:
Ah, I didn’t realize that. In that case I agree that we still need `atElifTypo`. Thanks for clarifying.